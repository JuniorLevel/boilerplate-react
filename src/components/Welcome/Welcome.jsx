// @flow
import { Button, Input, Flex, Layout, Typography } from 'antd';
import React from 'react';

const layoutStyle = {
	width: '100%',
	maxWidth: '300px',
	padding: '20px 15px',
};

interface IWelcomeProps {
	children: React.Node;
}

const Welcome = ({ children }: IWelcomeProps): React.Node => {
	const { Title } = Typography;

	return (
		<Flex justify="center" align="center" style={{ height: '100vh' }}>
			<Layout style={layoutStyle}>
				<Title style={{ textAlign: 'center' }}>Welcome</Title>
				<Input placeholder="email" style={{ marginBottom: '20px' }} />
				<Button block style={{ marginBottom: '20px' }}>
					Log in
				</Button>
				<Button block style={{ marginBottom: '20px' }}>
					create account
				</Button>
				{children}
			</Layout>
		</Flex>
	);
};

export default Welcome;
