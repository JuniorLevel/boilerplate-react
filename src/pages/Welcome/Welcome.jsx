// @flow
import { GoogleOutlined } from '@ant-design/icons';

import { Button, Input } from 'antd';
import React from 'react';
import styled from 'styled-components';

const Title = styled.h1`
	text-align: center;
	font-size: 3rem;
	margin-bottom: 50px;
	text-transform: uppercase;
	font-weight: 400;
`;

const Container = styled.div`
	background-color: #e4e1df;
	max-width: 250px;
	width: 100%;
	margin: 0 auto;
	padding: 1px 40px;
	margin-top: 25vh;
`;

const SignButtonPanel = styled.div`
	display: flex;
	justify-content: center;
	gap: 30px;
	align-items: center;
	margin-bottom: 60px;
`;

const BorderButton = styled.div`
	border: 5px solid black;
	border-radius: 50%;
	cursor: pointer;
`;

const EnterEmailInput = styled(Input)`
	border: 3px solid black;
	color: black;
	border-radius: 0px;
	font-size: 1rem;
	font-weight: bold;
	margin-bottom: 20px;
	text-align: center;
	&::placeholder {
		color: grey;
	}
`;

const Welcome = (): React$Node => (
	<Container>
		<Title>Welcome</Title>
		<EnterEmailInput placeholder="email" />
		<Button
			style={{
				border: '3px solid black',
				color: 'black',
				borderRadius: '0px',
				fontSize: '1rem',
				marginBottom: '20px',
				width: '100%',
				fontWeight: 700,
			}}
		>
			Log in
		</Button>
		<Button
			style={{
				border: '3px solid black',
				color: 'black',
				borderRadius: '0px',
				fontSize: '1rem',
				marginBottom: '70px',
				width: '100%',
				fontWeight: 700,
			}}
		>
			create account
		</Button>
		<SignButtonPanel>
			<BorderButton>
				<GoogleOutlined style={{ fontSize: '50px' }} />
			</BorderButton>
			<BorderButton>
				<span style={{ fontSize: '35px' }}>VK</span>
			</BorderButton>
		</SignButtonPanel>
	</Container>
);

export default Welcome;
