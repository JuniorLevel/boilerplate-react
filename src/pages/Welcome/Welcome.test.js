import { render, screen } from '@testing-library/react';
import React from 'react';
import Welcome from './Welcome';

describe('Welcome Component', () => {
	test('renders correctly', () => {
		render(<Welcome />);
		expect(screen.getByRole('button', { name: 'Log in' })).toBeInTheDocument();
	});
	test('takes snapshot', () => {
		const { asFragment } = render(<Welcome />);
		expect(asFragment()).toMatchSnapshot();
	});
});
