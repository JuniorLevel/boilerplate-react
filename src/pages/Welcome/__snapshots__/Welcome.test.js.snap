// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component takes snapshot 1`] = `
<DocumentFragment>
  <div
    class="Welcome__Container-sc-1bu4lih-1 jjMlgy"
  >
    <h1
      class="Welcome__Title-sc-1bu4lih-0 cteDnk"
    >
      Welcome
    </h1>
    <input
      class="ant-input css-dev-only-do-not-override-3rel02 ant-input-outlined Welcome__EnterEmailInput-sc-1bu4lih-4 ibOCEE"
      placeholder="email"
      type="text"
      value=""
    />
    <button
      class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
      style="border: 3px solid black; color: black; border-radius: 0px; font-size: 1rem; margin-bottom: 20px; width: 100%; font-weight: 700;"
      type="button"
    >
      <span>
        Log in
      </span>
    </button>
    <button
      class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
      style="border: 3px solid black; color: black; border-radius: 0px; font-size: 1rem; margin-bottom: 70px; width: 100%; font-weight: 700;"
      type="button"
    >
      <span>
        create account
      </span>
    </button>
    <div
      class="Welcome__SignButtonPanel-sc-1bu4lih-2 IgafW"
    >
      <div
        class="Welcome__BorderButton-sc-1bu4lih-3 bNhVyb"
      >
        <span
          aria-label="google"
          class="anticon anticon-google"
          role="img"
          style="font-size: 50px;"
        >
          <svg
            aria-hidden="true"
            data-icon="google"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M881 442.4H519.7v148.5h206.4c-8.9 48-35.9 88.6-76.6 115.8-34.4 23-78.3 36.6-129.9 36.6-99.9 0-184.4-67.5-214.6-158.2-7.6-23-12-47.6-12-72.9s4.4-49.9 12-72.9c30.3-90.6 114.8-158.1 214.7-158.1 56.3 0 106.8 19.4 146.6 57.4l110-110.1c-66.5-62-153.2-100-256.6-100-149.9 0-279.6 86-342.7 211.4-26 51.8-40.8 110.4-40.8 172.4S151 632.8 177 684.6C240.1 810 369.8 896 519.7 896c103.6 0 190.4-34.4 253.8-93 72.5-66.8 114.4-165.2 114.4-282.1 0-27.2-2.4-53.3-6.9-78.5z"
            />
          </svg>
        </span>
      </div>
      <div
        class="Welcome__BorderButton-sc-1bu4lih-3 bNhVyb"
      >
        <span
          style="font-size: 35px;"
        >
          VK
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
